src/com/github/kornilova203/matlab/MatlabSemanticEditorPosition.kt

```diff
@@ -0,0 +1,158 @@
+package com.github.kornilova203.matlab
+
+import com.github.kornilova203.matlab.psi.MatlabTypes
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import com.intellij.psi.tree.IElementType
+import com.intellij.util.text.CharArrayUtil
+
+class MatlabSemanticEditorPosition(editor: EditorEx, offset: Int) {
+    private var myIterator: HighlighterIterator? = null
```

Comment:
`myIterator` should not be nullable because it's initialised in constructor by non-null value.
Non-null assertions `!!` can be avoided in this file.
Also `init` block can be inlined:
```kotlin
    private var iterator = editor.highlighter.createIterator(offset)
    private var editor = editor
    private var chars = editor.document.charsSequence

```